services/file-retriever/src/services/dsnFetcher.ts

```diff
@@ -0,0 +1,99 @@
+import { FileResponse } from '../models/file.js'
+import {
+  blake3HashFromCid,
+  stringToCid,
+  decodeNode,
+  MetadataType,
+  cidToString,
+} from '@autonomys/auto-dag-data'
+import { PBNode } from '@ipld/dag-pb'
+import { env } from '../utils/env.js'
+import { HttpError } from '../http/middlewares/error.js'
+import { safeIPLDDecode } from '../utils/dagData.js'
+import mime from 'mime-types'
+
+const SUBSPACE_GATEWAY_URL = env('SUBSPACE_GATEWAY_URL')
+const MAX_SIMULTANEOUS_FETCHES = 10
+
+const fetchNode = async (cid: string): Promise<PBNode> => {
+  const objectMappingHash = blake3HashFromCid(stringToCid(cid))
+  const response = await fetch(
+    `${SUBSPACE_GATEWAY_URL}/objects/${objectMappingHash}`,
```

Comment:
This is the current url: `http://<IP>:<PORT>/data/{object_hash}`

Reply:
Fixed